script.js

```diff
@@ -1,4 +1,259 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
-};
+// /*
+// Included features:
+// - Last roll
+// - Winning %
+// - Most rolled
+// - Guessing +- 1
+// */
+
+/* 
+Fn:
+input: 
+- Number of users
+
+Each submit roll 2 dices
+track score
+*/
+var totalGames = 0;
+
+class player {
+  constructor(playername, dicenumber, score) {
+    this.playername = playername;
+    this.dicenumber = dicenumber;
+    this.combinedDiceNumber = this.combinedDiceNumber;
+    this.score = score;
+  }
+}
+
+function checkDiceForWinner() {
+  bblSortDice(plist);
+  plist[0].score += 1;
+}
+
+// BubbleSort directly to object.score highest to lowest (tbd)
+function bblSort(arr) {
+  for (var i = 0; i < arr.length; i++) {
+    // Last i elements are already in place
+    for (var j = 0; j < arr.length - i - 1; j++) {
+      // Checking if the item at present iteration
+      // is greater than the next iteration
+      if (arr[j].score > arr[j + 1].score) {
+        // If the condition is true then swap them
+        var temp = arr[j];
+        arr[j] = arr[j + 1];
+        arr[j + 1] = temp;
+      }
+    }
+  }
+}

```

Comment:
great work implementing bubble sort but this, along with your other 3 sorting functions below, feel very repetitive

to do simple sorting like in the case here, you can leverage on javascript's own array.sort() method

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort

Reply:
for numbers, you would have to supply a comparison function as an argument into sort

lets say i have an array `var arr = [20, 10, 30, 40]`

```
arr.sort(function(a, b) {
  return a - b;
});
console.log(arr); // [10, 20, 30, 40]
```
basically, the comparison function takes in two arguments (which we call a and b) then it applies this sorting function to every pair of elements in the array. if the comparison function returns < 0, then .sort will place a before b, and b before a if the comparison function returns > 0

that's why when we return `a - b`, it has the effect of sorting it in ascending order. lets just take the first two elements as example, a will be 20 and b will be 10, so `a - b` will be 10 > 0, so 10 will be placed before 20

to sort in descending order, we just need to return `b - a`

Reply:
lmk if you got any qns on this! i know it can be a complicated concept to grasp at first